[2025-04-12T14:36:00.661+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-12T14:36:00.748+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sparktestdag.spark_submit_job manual__2025-04-12T14:35:43.739921+00:00 [queued]>
[2025-04-12T14:36:00.779+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sparktestdag.spark_submit_job manual__2025-04-12T14:35:43.739921+00:00 [queued]>
[2025-04-12T14:36:00.779+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-04-12T14:36:00.942+0000] {taskinstance.py:2890} INFO - Executing <Task(SparkSubmitOperator): spark_submit_job> on 2025-04-12 14:35:43.739921+00:00
[2025-04-12T14:36:00.950+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=288) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-04-12T14:36:00.953+0000] {standard_task_runner.py:72} INFO - Started process 290 to run task
[2025-04-12T14:36:00.960+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'sparktestdag', 'spark_submit_job', 'manual__2025-04-12T14:35:43.739921+00:00', '--job-id', '46', '--raw', '--subdir', 'DAGS_FOLDER/sparktestdag.py', '--cfg-path', '/tmp/tmps4wsr2m_']
[2025-04-12T14:36:00.962+0000] {standard_task_runner.py:105} INFO - Job 46: Subtask spark_submit_job
[2025-04-12T14:36:01.114+0000] {task_command.py:467} INFO - Running <TaskInstance: sparktestdag.spark_submit_job manual__2025-04-12T14:35:43.739921+00:00 [running]> on host 2da13769e137
[2025-04-12T14:36:01.206+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='sparktestdag' AIRFLOW_CTX_TASK_ID='spark_submit_job' AIRFLOW_CTX_EXECUTION_DATE='2025-04-12T14:35:43.739921+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-12T14:35:43.739921+00:00'
[2025-04-12T14:36:01.210+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-04-12T14:36:01.211+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-04-12T14:36:01.212+0000] {logging_mixin.py:190} INFO - Current task name:spark_submit_job state:running start_date:2025-04-12 14:36:00.759314+00:00
[2025-04-12T14:36:01.213+0000] {logging_mixin.py:190} INFO - Dag name:sparktestdag and current dag run status:running
[2025-04-12T14:36:01.214+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-04-12T14:36:01.226+0000] {base.py:84} INFO - Retrieving connection 'spark_master'
[2025-04-12T14:36:01.228+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://test-spark-1:7077 --name arrow-spark --deploy-mode client /pspark/testspark.py
[2025-04-12T14:36:01.355+0000] {spark_submit.py:645} INFO - /home/***/.local/lib/python3.12/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-04-12T14:36:01.367+0000] {spark_submit.py:645} INFO - /home/***/.local/lib/python3.12/site-packages/pyspark/bin/spark-class: line 71: /usr/lib/jvm/java-17-openjdk-arm64/bin/java: No such file or directory
[2025-04-12T14:36:01.368+0000] {spark_submit.py:645} INFO - /home/***/.local/lib/python3.12/site-packages/pyspark/bin/spark-class: line 97: CMD: bad array subscript
[2025-04-12T14:36:01.396+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 196, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://test-spark-1:7077 --name arrow-spark --deploy-mode client /pspark/testspark.py. Error code is: 1.
[2025-04-12T14:36:01.417+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2025-04-12T14:36:01.418+0000] {logging_mixin.py:190} INFO - Task start:2025-04-12 14:36:00.759314+00:00 end:2025-04-12 14:36:01.416403+00:00 duration:0.657089
[2025-04-12T14:36:01.420+0000] {logging_mixin.py:190} INFO - Task:<Task(SparkSubmitOperator): spark_submit_job> dag:<DAG: sparktestdag> dagrun:<DagRun sparktestdag @ 2025-04-12 14:35:43.739921+00:00: manual__2025-04-12T14:35:43.739921+00:00, state:running, queued_at: 2025-04-12 14:35:43.927897+00:00. externally triggered: True>
[2025-04-12T14:36:01.421+0000] {logging_mixin.py:190} INFO - Failure caused by Cannot execute: spark-submit --master spark://test-spark-1:7077 --name arrow-spark --deploy-mode client /pspark/testspark.py. Error code is: 1.
[2025-04-12T14:36:01.422+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=sparktestdag, task_id=spark_submit_job, run_id=manual__2025-04-12T14:35:43.739921+00:00, execution_date=20250412T143543, start_date=20250412T143600, end_date=20250412T143601
[2025-04-12T14:36:01.489+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-04-12T14:36:01.494+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 46 for task spark_submit_job (Cannot execute: spark-submit --master spark://test-spark-1:7077 --name arrow-spark --deploy-mode client /pspark/testspark.py. Error code is: 1.; 290)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 196, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://test-spark-1:7077 --name arrow-spark --deploy-mode client /pspark/testspark.py. Error code is: 1.
[2025-04-12T14:36:01.534+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-04-12T14:36:01.556+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-12T14:36:01.561+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
